### ISSUE TITLE
kube-proxy fails to to install iptables KUBE-PORTALS-CONTAINER rule for service

### ISSUE THREAD

**Issue by rpatrick00** [kubernetes/kubernetes#11587]
I have Kubernetes 0.17 configured locally using flannel and everything had been working perfectly until today (I deleted a few keys from etcd that I thought were no longer needed due to me switching my Kubernetes artifact naming scheme).  Now, kube-proxy is no longer working and I am getting connection refused when trying to access the Service using its public IP and port.

I see this in the system log but I was seeing these before even when it was working so I am not sure if the two are related:

```
Jul 20 07:11:07 slc06ptm kube-proxy: E0720 07:11:07.117205    1410 proxier.go:591] Failed to install iptables KUBE-PORTALS-CONTAINER rule for service "testme/auction-test:"
Jul 20 07:11:07 slc06ptm kube-proxy: E0720 07:11:07.117252    1410 proxier.go:390] Failed to ensure portal for "testme/auction-test:": error checking rule: exit status 2: iptables v1.4.21: host/network `<nil>' not found
Jul 20 07:11:07 slc06ptm kube-proxy: Try `iptables -h' or 'iptables --help' for more information.
```

Note that I have the docker daemon running with the --iptables=false flag and flanneld running with the --ip-masq flag.  Given that docker0 is layering on top of flannel0, I assume this is ok and should not be a problem for kube-proxy, which I assume is using docker0 to communicate with the containers.

Any help would be greatly appreciated!

Thanks,
Robert

**Comment by mikedanese** [comment]
What keys did you delete? It's not recommended to remove anything in /registry using the etcd api dierctly. You should use the apiserver and kubectl. Is it possible for you to upgrade?

**Comment by rpatrick00** [comment]
Thanks for your reply, I found out what was causing my problem,  in my
application that is using the REST API, I was specifying the publicIPs of
the service using DNS names while in my kubectl script , that I was using
for testing to create the service, I was actually using IP addresses.

When I switched my kubectl scripts to DNS, they failed too.  When I
switched my app to use IP Addresses, it started working properly.  I am not
sure that I understand why using DNS names instead of the underlying IP
addresses don't work but at least it is working now...
